pyrefly/lib/alt/class/class_metadata.rs

```diff
@@ -1242,4 +1259,142 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
             .collect();
         ClassMro::new(cls, bases_with_mros, errors)
     }
+
+    /// Check if a class inherits from abc.ABC
+    fn check_inherits_from_abc(
+        &self,
+        _cls: &Class,
+        bases_with_metadata: &Vec<(Class, Arc<ClassMetadata>)>,
+    ) -> bool {
+        // Check if any base is abc.ABC
+        for (base, _) in bases_with_metadata {
+            if base.module_name().as_str() == "abc" && base.name().as_str() == "ABC" {
+                return true;
+            }
+        }
+
+        // Check if any base inherits from abc.ABC
+        for (_, metadata) in bases_with_metadata {
+            if metadata.inherits_from_abc() {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    /// Collect unimplemented abstract methods for a class
+    fn collect_unimplemented_abstract_methods(
+        &self,
+        cls: &Class,
+        bases_with_metadata: &Vec<(Class, Arc<ClassMetadata>)>,
+    ) -> Vec<Name> {
+        use starlark_map::small_set::SmallSet;
+
+        let mut abstract_methods = SmallSet::new();
+
+        // Collect abstract methods from all base classes
+        for (_, metadata) in bases_with_metadata {
+            for method in metadata.abstract_methods() {
+                abstract_methods.insert(method.clone());
```

Comment:
q here: is it possible that one base class declares a method as abstract but another base class provides the concrete impl? if so, idt we handle that case.